src/LiquidStaking/LiquidStaking.cpp

```diff
@@ -178,6 +194,51 @@ Proto::Output Builder::buildTortuga() const {
     return output;
 }
 
+Proto::Output Builder::buildLidoEVM() const {
+    Proto::Output output;
+    if (!mSmartContractAddress) {
+        *output.mutable_status() = generateError(Proto::ERROR_SMART_CONTRACT_ADDRESS_NOT_SET, "Lido protocol require the smart contract address to be set");
+        return output;
+    }
+    auto input = Ethereum::Proto::SigningInput();
+
+    if (this->mBlockchain == Proto::ETHEREUM) {
+        input.set_tx_mode(Ethereum::Proto::Enveloped);
+    }
+    input.set_to_address(*mSmartContractAddress);
+
+    auto& transfer = *input.mutable_transaction()->mutable_contract_generic();
+
+
+    auto visitFunctor = [&transfer, this, &output](const TAction& value) {
+        Data payload;
+        uint256_t amount;
+
+        if (auto* stake = std::get_if<Proto::Stake>(&value); stake) {
+            internal::handleStake(*stake, mBlockchain, payload, amount, Proto::Lido);
+        } else {
+            *output.mutable_status() = generateError(Proto::ERROR_OPERATION_NOT_SUPPORTED_BY_PROTOCOL, "Lido protocol only support stake action for now");
```

Comment:
Do we need to return here to avoid unnecessary `internal::setTransferDataAndAmount(transfer, payload, amount);` call

Reply:
let's refactor when moving to rust, anyway app should always check the status of the output, thanks for the note I keep it in mind